cinema/views.py

```diff
@@ -56,3 +78,54 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset.prefetch_related("movie", "cinema_hall")
+        if self.action == "list":
+            queryset = queryset.annotate(
+                tickets_count=Count("tickets")
+            )
+
+            for item in queryset:
+                item.tickets_available = (item.cinema_hall.capacity
+                                          - item.tickets_count)
```

Comment:
Avoid performing an iteration over all items in the queryset, think of ways to implement all logic using Django ORM.